Reorganize windowmaker dockapps

AndersonTorres opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

SuperSandro2000 commented on pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix:
```diff
@@ -0,0 +1,30 @@
+
+  src = dockapps-sources;
+
+  nativeBuildInputs = [ pkgconfig ];
```
```suggestion
  nativeBuildInputs = [ pkg-config ];
```

SuperSandro2000 commented on pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix:
```diff
@@ -0,0 +1,30 @@
+{ stdenv, dockapps-sources, pkgconfig, libX11, libXpm, libXext, alsaLib }:
```
```suggestion
{ stdenv, dockapps-sources, pkg-config, libX11, libXpm, libXext, alsaLib }:
```

SuperSandro2000 commented on pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix:
```diff
@@ -0,0 +1,30 @@
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ libX11 libXpm libXext alsaLib ];
+
+  postUnpack = "sourceRoot=\${sourceRoot}/${pname}";
```
```suggestion
  postUnpack = ''
    sourceRoot=\${sourceRoot}/${pname}
  '';
```

SuperSandro2000 commented on pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix:
```diff
@@ -0,0 +1,30 @@
+    runHook preInstall
+    install -d $out/bin
+    install -t $out/bin AlsaMixer.app
+    (cd $out/bin; ln -s AlsaMixer.app AlsaMixer)
```
```suggestion
    ln -s $out/bin/AlsaMixer.app $out/bin/AlsaMixer
```

SuperSandro2000 commented on pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix:
```diff
@@ -0,0 +1,30 @@
+    description = "Alsa mixer application for Windowmaker";
+    homepage = "https://www.dockapps.net/alsamixerapp";
+    license = licenses.gpl2Plus;
+    maintainers = [ maintainers.bstrik ];
```
```suggestion
    maintainers = with maintainers; [ bstrik ];
```

SuperSandro2000 commented on pkgs/applications/window-managers/windowmaker/dockapps/default.nix:
```diff
@@ -0,0 +1,21 @@
+
+  dockapps-sources = pkgs.fetchgit {
+    url = "https://repo.or.cz/dockapps.git";
+    rev = "b2b9d872ee61c9b329e4597c301e4417cbd9c3ea";
```
Is there a branch or tag we could track?

AndersonTorres replied:
Not so easily. It is essentially a monorepo of a plethora of dockapps.

SuperSandro2000 commented on pkgs/applications/window-managers/windowmaker/dockapps/libdockapp.nix:
```diff
@@ -0,0 +1,26 @@
+  postUnpack = "sourceRoot=\${sourceRoot}/${pname}";
+
+  # There is a bug on --with-font
+  configureFlags = [ "--with-examples=no" "--with-font=no" ];
```
```suggestion
  configureFlags = [
    "--with-examples=no"
    "--with-font=no"
  ];
```

SuperSandro2000 commented on pkgs/applications/window-managers/windowmaker/dockapps/libdockapp.nix:
```diff
@@ -0,0 +1,26 @@
+    description = "A library providing a framework for dockapps";
+    homepage = "https://www.dockapps.net/libdockapp";
+    license = licenses.gpl2Plus;
+    maintainers = [ maintainers.bstrik ];
```
```suggestion
    maintainers = with maintainers; [ bstrik ];
```

SuperSandro2000 commented on pkgs/applications/window-managers/windowmaker/dockapps/wmCalClock.nix:
```diff
@@ -0,0 +1,35 @@
+  postUnpack = "sourceRoot=\${sourceRoot}/${pname}/Src";
+
+  preBuild = ''
+    makeFlagsArray+=( 
+      CC="cc"
+      INCDIR="-I${libX11.dev}/include -I${libXext.dev}/include -I${libXpm.dev}/include"
+      LIBDIR="-I${libX11}/lib -I${libXext}/lib -I${libXpm}/lib" )
```
```suggestion
    makeFlagsArray+=( 
      CC="cc"
      INCDIR="-I${libX11.dev}/include -I${libXext.dev}/include -I${libXpm.dev}/include"
      LIBDIR="-I${libX11}/lib -I${libXext}/lib -I${libXpm}/lib"
    )
```

SuperSandro2000 commented on pkgs/applications/window-managers/windowmaker/dockapps/wmsystemtray.nix:
```diff
@@ -1,20 +1,22 @@
   src = fetchurl {
-     url = "mirror://sourceforge/project/wmsystemtray/wmsystemtray/wmsystemtray-1.4.tar.gz";
-     sha256 = "8edef43691e9fff071000e29166c7c1ad420c0956e9068151061e881c8ac97e9";
+    url = "mirror://sourceforge/project/${pname}/${pname}/${pname}-${version}.tar.gz";
+     sha256 = "sha256-jt70NpHp//BxAA4pFmx8GtQgwJVukGgVEGHogcisl+k=";
```
```suggestion
    url = "mirror://sourceforge/project/${pname}/${pname}/${pname}-${version}.tar.gz";
    sha256 = "jt70NpHp//BxAA4pFmx8GtQgwJVukGgVEGHogcisl+k=";
```

AndersonTorres replied:
Well, this is the format spit out by the command when I use it via fakeSha256 trick.

AndersonTorres committed:
Dockapps: new structure

AndersonTorres committed:
Dockapps: fetch sources

Most of dockapps live in a monorepo maintained by dockapps.net. Unfortunately
there is not an obvious way to fix a particular release, tag or whatever from
this monorepo.

AndersonTorres committed:
libdockapp: init at 0.7.3

AndersonTorres committed:
AlsaMixer-app: stay at 0.2.1

AndersonTorres committed:
wmCalClock: stay at 1.25

AndersonTorres committed:
wmsm-app: stay at 0.2.1

AndersonTorres committed:
wmsystemtray: stay at 1.4

AndersonTorres committed:
Dockapps: cleanup
